tests/e2e-agents/e2e_agents/infra/wait_for_ready.py

```
@@ -0,0 +1,50 @@
+
+def wait_for_app(url: str, timeout: int = 180) -> bool:
+    """Block until the app at url responds with any 2xx/3xx, or timeout.
+
+    Returns True if healthy, False if timed out.
+    """
+    start = time.time()
+    interval = 2.0
+
+    while time.time() - start < timeout:
+        try:
+            req = urllib.request.Request(url, method="GET")
```

Ruff (0.15.12)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)